x-pack/plugins/data_enhanced/public/search/search_abort_controller.ts

```diff
@@ -18,11 +18,7 @@ export class SearchAbortController {
   private destroyed = false;
   private reason?: AbortReason;
 
-  constructor(abortSignal?: AbortSignal, timeout?: number) {
-    if (abortSignal) {
-      this.addAbortSignal(abortSignal);
-    }
-
+  constructor(timeout?: number) {
```

Comment:
Not visible in diff of this pr, but could you please double-check if the order here is correct:
https://github.com/elastic/kibana/blob/7e394d4d6f94c557749df5ce828f81d0ff988c96/x-pack/plugins/data_enhanced/public/search/search_abort_controller.ts#L67-L70

Should it be first `.abort()` and then `cleanup()` ? Otherwise external signals won't get notified of `abort`?

Reply:
`cleanup` doesn't do anything related to `abortController`, so I don't think there is any connection between the two.
What do you think?